backend/app.js

```diff
@@ -197,6 +197,177 @@ app.post("/api/createEvent", authenticateToken, async (req, res) => {
+  const pool = await getDbConnection(); 
+  const connection = await pool.getConnection();
+  await connection.beginTransaction();
+
+  try {
+
+    for (let i = 0; i < infoArray.length; i++)
+    {
+      const {day, start, end} = infoArray[i];
+
+      // validate inputs
+      // Convert dates and times to Date objects for comparison
+      const startTime = new Date(`${infoArray[i].day}T${infoArray[i].start}`);
+      const endTime = new Date(`${infoArray[i].day}T${infoArray[i].end}`);
+
+
+      // Check that end time is not before start time
+      if (endTime < startTime)
+      {
+        throw new Error(`Availability ${i + 1}: End time cannot be before start time!`);
+      }
+
+
+      //check for duplicate entries- duplicated being entries with the identical UserID, Date, startTime and endTimes;
+      const [duplicates] = await connection.query(
+      `SELECT * FROM Availability
+      WHERE UserID = ? AND Date = ? 
+      AND StartTime = ? AND EndTime = ?`,
+      [userID, day, start, end]
+      );
+
+
+      if (duplicates.length > 0) 
+      {
+        throw new Error(`Availability ${i + 1}: is a duplicate`);
+      }
+      
+
+      //generate ID
+      const availabilityID = uuid();
+
+
+      //create inserts
+      const [result] = await connection.query("INSERT INTO Availability (AvailabilityID, UserID, EventID, Date, StartTime, EndTime) VALUES (?, ?, ?, ?, ?, ?)",
+        [
+          availabilityID,
+          userID,
+          eventID,
+          day,
+          start,
+          end,
+        ]);
+
+    }
+
+  // apply inserts to database
+  await connection.commit();  
+
+
+  // Respond with success message
+  const returnStats = {
+    message :"All availabilities successfully inserted.",
+    status : 201
+  };
+  connection.release();
+  return returnStats;
+  } 
+  catch (error) {
+    //avoid committing messages
+    await connection.rollback();
+
+
+    //response 
+    const returnStats = {
+    message : error.message,
+    status : 400
+    };
+
+    //direct connection needs release (end), unlike pools
+    connection.release();
+    return returnStats;
+
+  }
+}
+
+
+async function deleteAvailability(eventID, userID)
+{
+  // get connection to database
+  const pool = await getDbConnection(); 
+  const connection = await pool.getConnection();
+  await connection.beginTransaction();
+
+  //only one query to delete all 
+  const [result] =  await connection.query("DELETE FROM Availability WHERE EventID = ? AND UserID = ?",
+    [eventID, userID]);
+
+
+  const returnStats = {
+    message :"All availabilities successfully deleted.",
+    status : 201
+  };
+
+  connection.release();
+  return returnStats;
+
+}
+
+
+ app.post("/api/updateAvailability", authenticateToken, async (req, res) => {
+  const {eventID, availability} = req.body;
+  const userID = req.user.userId;
+  
+
+  //get all the date with start/end times
+    const infoArray = req.body.availability.flatMap(slot =>
```

Comment:
Change 
```suggestion
    const availabilitySlots = req.body.availability.flatMap(slot =>
```

infoArray doesn't really make sense